### ISSUE TITLE
argmax and argmin ops return the random number for scalar tensor when specifying dim

### ISSUE THREAD

**Issue by stu1130** [pytorch/pytorch#37084]
## 🐛 Bug

argmax and argmin ops return the random number for scalar tensor when specifying dim
 
## To Reproduce
```
>>> import torch
>>> torch.__version__
'1.5.0'
>>> a = torch.tensor(5.)
>>> torch.argmax(a) # it gives the right result
tensor(0)
>>> torch.argmax(a, dim=0)
tensor(140457834750256)
```

cc @ezyang @gchanan @zou3519

**Comment by mruberry** [comment]
Thanks for the bug report, @stu1130! This looks like a regression from 1.4.

**Comment by emcastillo** [comment]
I reproduced the bug and realized that this only happens when the tensor is a scalar.
`a = torch.tensor([5.])` seems to be working fine. There might be some problem in the reduction routines when dealing with 0-dim tensors

**Comment by ngimel** [comment]
If we had tests for reduction on scalars ASAN test should have caught it. #37214 adds tests for scalar argmax, probably it makes sense to beef up testing for scalars for more operations.
